src/language/concurrency.md

```diff
@@ -226,314 +280,195 @@ using asynchronous operations as necessary.
-run directly in the main isolate.
-Using `Isolate.run()` to run it instead enables concurrency.
-The worker isolate completely abstracts the computations
-of `_readAndParseJson()`. It can complete without blocking the main isolate.
+### Using isolates
 
-The result of `Isolate.run()` is always a Future,
-because code in the main isolate continues to run.
-Whether the computation the worker isolate executes
-is synchronous or asynchronous doesn't impact the
-main isolate, because it's running concurrently either way.
+There are two ways to work with isolates in Dart, depending on the use-case:
 
-For the complete program, check out the [send_and_receive.dart][] sample.
+* Use [`Isolate.run()`][] to perform a single computation on a separate thread.
+* Use [`Isolate.spawn()`][] to create an isolate that will handle
+  multiple messages over time, or a background worker.
 
-{% comment %}
-TODO:
-Should create a diagram for the current example.
-Previous example's diagram and text for reference:
+In most cases, `Isolate.run` is the recommended
+API to run processes in the background.
 
-  The following figure illustrates the communication between
-  the main isolate and the worker isolate:
-  
-  ![A figure showing the previous snippets of code running in the main isolate and in the worker isolate](/assets/img/language/concurrency/isolate-api.png)
-{% endcomment %}
+#### `Isolate.run()`
 
-#### Sending closures with isolates
+The static `Isolate.run()` method requires one argument: a callback that will be
+run on the newly spawned isolate.
 
-You can also create a simple worker isolate with `run()` using a
-function literal, or closure, directly in the main isolate.
-
-<?code-excerpt "lib/simple_isolate_closure.dart (main)"?>
+<?code-excerpt "lib/isolate_run_syntax.dart"?>
 ```dart
-const String filename = 'with_keys.json';
-
-void main() async {
-  // Read some data.
-  final jsonData = await Isolate.run(() async {
-    final fileData = await File(filename).readAsString();
-    final jsonData = jsonDecode(fileData) as Map<String, dynamic>;
-    return jsonData;
-  });
+int slowFib(int n) => n <= 1 ? 1 : slowFib(n - 1) + slowFib(n - 2);
 
-  // Use that data.
-  print('Number of JSON keys: ${jsonData.length}');
+// Compute without blocking current isolate.
+void fib40() async {
+  var result = await Isolate.run(() => slowFib(40));
+  print('Fib(40) = $result');
 }
 ```
 
-This example accomplishes the same as the previous.
-A new isolate spawns, computes something, and sends back the result.
-
-However, now the isolate sends a [closure][].
-Closures are less limited than typical named functions,
-both in how they function and how they're written into the code.
-In this example, `Isolate.run()` executes what looks like local code, concurrently.
-In that sense, you can imagine `run()` to work like a control flow operator
-for "run in parallel".
-
-[closure]: /language/functions#anonymous-functions
-
-### Sending multiple messages between isolates
```

Comment:
This section is removed until the next PR. Per suggestion by @parlough  (which I agree with), I'll want to reference the isolate example from the new isolates page rather than include it on both pages. So this will need to be added once there's something to reference.